setuptools/dist.py

```diff
@@ -565,6 +572,31 @@ def _clean_req(self, req):
         req.marker = None
         return req
 
+    def _finalize_license_files(self):
+        """Compute names of all license files which should be included."""
+        files = set()
+        license_files: Optional[List[str]] = self.metadata.license_files
+        patterns: Set[str] = set(license_files) if license_files else set()
+
+        license_file: Optional[str] = self.metadata.license_file
+        if license_file:
+            patterns.add(license_file)
+
+        if license_files is None and license_file is None:
+            # Default patterns match the ones wheel uses
+            # See https://wheel.readthedocs.io/en/stable/user_guide.html
+            # -> 'Including license files in the generated wheel file'
+            patterns = ('LICEN[CS]E*', 'COPYING*', 'NOTICE*', 'AUTHORS*')
+
+        for pattern in patterns:
+            for path in iglob(pattern):
+                if path.endswith('~'):
+                    continue
+                if path not in files and os.path.isfile(path):
+                    files.add(path)
+
+        self.metadata.license_files_computed = sorted(files)
```

Comment:
Can we retain order of license files rather than enforcing order? For the manifest, that's less important, but for metadata, it may prove useful to honor the user's supplied order.

Reply:
The user only provides patterns which are used for `glob`. As such, I don't think retaining any order would make much sense.

Reply:
Alright. I'll probably re-write it to avoid sorting or at least avoid sorting user-supplied values.